src/phoenix/core/traces.py

```diff
@@ -152,34 +150,39 @@ def get_spans(
         stop_time: Optional[datetime] = None,
         root_spans_only: Optional[bool] = False,
         span_ids: Optional[Iterable[SpanID]] = None,
+        project_name: Optional[str] = None,
     ) -> Iterator[Span]:
-        if not self._spans:
-            return
-        min_start_time, max_stop_time = cast(
-            Tuple[datetime, datetime],
-            self.right_open_time_range,
-        )
-        start_time = start_time or min_start_time
-        stop_time = stop_time or max_stop_time
-        if span_ids is not None:
-            with self._lock:
+        if not project_name:
+            project_name = _DEFAULT_PROJECT_NAME
+        with self._lock:
+            if not (project := self._projects.get(project_name)):
+                return
+            min_start_time, max_stop_time = cast(
+                Tuple[datetime, datetime],
+                self.right_open_time_range(project_name),
+            )
+            start_time = start_time or min_start_time
+            stop_time = stop_time or max_stop_time
+            if span_ids is not None:
                 spans = tuple(
                     span
                     for span_id in span_ids
                     if (
-                        (span := self._spans.get(span_id))
+                        (span := project.spans.get(span_id))
                         and start_time <= span.start_time < stop_time
                         and (not root_spans_only or span.parent_id is None)
                     )
                 )
-        else:
-            sorted_spans = (
-                self._start_time_sorted_root_spans
-                if root_spans_only
-                else self._start_time_sorted_spans
-            )
-            # make a copy because source data can mutate during iteration
-            with self._lock:
+            else:
+                project = self._projects.get(project_name)
```

Comment:
similar to above, tracing having projects rather than a project having traces feels backwards to me

Reply:
discussed offline